### PR TITLE
[Cart][Behat] Scenarios for visitor's cart taxes

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets |
| License       | MIT
| Doc PR        |

I've closed https://github.com/Sylius/Sylius/pull/4045 with the same changes, as it will be easier to me (with rebase) and for everybody else (with waiting for "Awaiting Review") ;)
This is the last episode of "Taxes and shipment scenarios" series, hope you'll enjoy it.

### CODE REVIEW

File: features/shop/apply_correct_taxes_on_visitor_cart.feature [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        And store allows paying offline
+
+    Scenario: Proper taxes for taxed product
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "€123.00"
+        And my cart taxes should be "€23.00"
+
+    Scenario: Proper taxes after specifying shipping address
+        When I add product "PHP T-Shirt" to the cart
+        And I proceed logging as "john@example.com" guest with "Australia" as shipping country
```
Comment (pjedrzejewski) [inline features/shop/apply_correct_taxes_on_visitor_cart.feature]: Not really logging in, how about ``I proceed as guest``?

Comment (Zales0123) [inline features/shop/apply_correct_taxes_on_visitor_cart.feature]: :+1:

File: features/shop/apply_correct_taxes_on_visitor_cart.feature [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        And store allows paying offline
+
+    Scenario: Proper taxes for taxed product
+        When I add product "PHP T-Shirt" to the cart
+        Then my cart total should be "€123.00"
+        And my cart taxes should be "€23.00"
+
+    Scenario: Proper taxes after specifying shipping address
+        When I add product "PHP T-Shirt" to the cart
+        And I proceed as guest "john@example.com" with "Australia" as shipping country
```
Comment (michalmarcinkowski) [inline features/shop/apply_correct_taxes_on_visitor_cart.feature]:
```
Given I have product ...
When I proceed as ...
Then ...
```

File: src/Sylius/Behat/Context/Setup/ShippingContext.php [hunk 3]
```diff
@@ -12,6 +12,7 @@
 namespace Sylius\Behat\Context\Setup;
 
 use Behat\Behat\Context\Context;
+use Behat\Gherkin\Node\TableNode;
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Setup/ShippingContext.php]: Do we need this?

File: features/shop/apply_correct_taxes_on_visitor_cart.feature [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        And there is "EU" zone containing all members of European Union
+        And there is rest of the world zone containing all other countries
+        And default currency is "EUR"
+        And default tax zone is "EU"
+        And the store has "EU VAT" tax rate of 23% for "Clothes" within "EU" zone
+        And the store has "EU VAT" tax rate of 10% for "Clothes" for the rest of the world
+        And the store has a product "PHP T-Shirt" priced at "€100.00"
+        And product "PHP T-Shirt" belongs to "Clothes" tax category
+        And the store ships everything for free to all available locations
+        And the store allows paying offline
```
Comment (michalmarcinkowski) [inline features/shop/apply_correct_taxes_on_visitor_cart.feature]: The two steps above will not be needed after rebase ;)

**Comment by pamil** [comment]
:+1:

**Comment by michalmarcinkowski** [comment]
Thanks Mateusz! :+1:

**Pull request merged**